File: typescript/rna-transcription/rna-transcription.ts
export default class Transcriptor{
  
  toRna(input: string){
    
    let rnaMap:{[key:string]: string} = {
      'C': 'G',
      'G': 'C',
      'A': 'U',
      'T': 'A'
    };

    let output = '';
    for (var index = 0; index < input.length; index++) {
      let c = rnaMap[input[index]];
      if(c === undefined){
        throw "Invalid input DNA.";
      }
      output += c;
    }

    return output;
  }

}

console.log((new Transcriptor).toRna('ACGTGGTCTTAA'))

File: python/rna-transcription/rna_transcription.py
def to_rna(dna):
    dna2rna_map = {
        'G': 'C',
        'C': 'G',
        'T': 'A',
        'A': 'U', 
    }

    rna = ''
    for c in dna:
        try:
            rna += dna2rna_map[c]
        except KeyError:
            return ''
    return rna
